src/store/index.ts

```diff
@@ -61,10 +62,8 @@ export const getPersistedState = () => {
 
 const hydrationReducer: typeof rootReducer = (state, action) => {
   if (action.type === HYDRATE_ACTION) {
-    return {
-      ...state,
-      ...action.payload,
-    }
+    // `merge` mutates the first argument, so we need to create a new object
+    return merge({}, state, action.payload)
```

Comment:
Merging is walking on thin ice, what if the stored data is in conflict with the new data?

Can you explain a little bit more about how new data may appear in the state before the hydration and why just avoiding setting undefined is not enough to fix the problem?

Reply:
It's related to [this comment](https://github.com/safe-global/web-core/pull/1527#discussion_r1071025396).

If we extend the intial state of a slice, the default values are currently [overwritten by the persisted state](https://github.com/safe-global/web-core/blob/dev/src/store/index.ts#L63-L68). We will alternatively have to manually assign "extensions" of initial state in dispatched actions if they are not present.

Reply:
I'm aware, but the problem seems to revolve around "undefined" being retrieved from the LS.

Reply:
After discussing this and confirming that is **only** related to new initial state properties, I've added an explanatory comment and a further test. Lmk what you think.